lib/astarte_vmq_plugin.ex

```diff
@@ -112,6 +105,19 @@ defmodule Astarte.VMQ.Plugin do
     end
   end
 
+  def delete_client(client_id) do
+    mountpoint = ''
```

Comment:
How does this differ from the function just above? The only difference is that you're calling `ack_device_deletion`, but then again why does that call need to be nested instead of serial, e.g. at the callsite
```elixir
# In handle_rpc
with :ok <- Plugin.disconnect_client(client_id, true),
     :ok <- Plugin.ack_device_deletion(client_id) do
  :ok
else
  {:error, reason} ->
    ...
end
```
This additionally allows you to reply with an error if the query in `ack_device_deletion` fails instead of crashing the whole RPC AMQP consumer like it would happen in the current version